code/game/machinery/autodoc.dm

```diff
@@ -630,16 +655,45 @@
 		update_use_power(2)
 		occupant = usr
 		icon_state = "autodoc_closed"
+		var/implants = list(/obj/item/implant/chem, /obj/item/implant/death_alarm, /obj/item/implant/loyalty, /obj/item/implant/tracking, /obj/item/implant/neurostim)
+		var/mob/living/carbon/human/H = occupant
+		var/doc_dat
+		med_scan(H, doc_dat, implants, TRUE)
 		start_processing()
 		connected.start_processing()
 
 		for(var/obj/O in src)
 			cdel(O)
 		add_fingerprint(usr)
 
-/obj/machinery/autodoc/proc/go_out()
-	if(!occupant) return
+/obj/machinery/autodoc/proc/go_out(notice_code = FALSE)
+	if(!occupant)
+		return
 	occupant.forceMove(loc)
+	if(connected.release_notice) //If auto-release notices are on as they should be, let the doctors know what's up
+		var/reason = "Reason for discharge: Procedural completion."
+		switch(notice_code)
+			if(1)
```

Comment:
Magic numbers, use defines.

Reply:
?

You want me to use defines for the notice code?

Really? A separate define for each event type?

Reply:
Yes, that's the idea. Unless @Qustinnus disagrees here?

Reply:
@Surrealaser yes because i cant be arsed finding out what those magic numbers mean

Reply:
its essentialy an enum now but without the identifier which makes me fucking weep every time i review someones code and I have to extrapolate what the fuck 1 to 17 means.